src/main/java/de/tum/in/www1/artemis/service/iris/IrisGPT3_5Service.java

```diff
@@ -0,0 +1,202 @@
+import org.springframework.http.MediaType;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import de.tum.in.www1.artemis.domain.iris.IrisMessage;
+import de.tum.in.www1.artemis.domain.iris.IrisMessageContent;
+import de.tum.in.www1.artemis.domain.iris.IrisMessageSender;
+import de.tum.in.www1.artemis.domain.iris.IrisSession;
+
+@Service
+public class IrisGPT3_5Service implements IrisModel {
+
+    private final Logger log = LoggerFactory.getLogger(IrisGPT3_5Service.class);
+
+    /**
+     * The RestTemplate used to send requests to the LLM API.
+     * TODO: Needs to be configured?
+     */
+    private final RestTemplate restTemplate;
+
+    /**
+     * The URL to send the API requests to.
+     */
+    @Value("${artemis.iris.models.gpt3_5.url}")
+    private URL apiURL;
+    // https://ase-eu01.openai.azure.com/openai/deployments/gpt-35/chat/completions?api-version=2023-03-15-preview
+
+    /**
+     * The API key to use when sending requests to the API.
+     */
+    @Value("${artemis.iris.models.gpt3_5.api-key}")
+    private String apiKey;
+
+    /**
+     * The ID of the model to request from the API.
+     */
+    @Value("${artemis.iris.models.gpt3_5.model}")
+    private String model;
+
+    /**
+     * The temperature parameter controls the randomness of the model. Lower temperatures make the model more
+     * deterministic, while higher temperatures make the model more varied in its responses. For a more detailed
+     * explanation, see <a href="https://platform.openai.com/docs/api-reference/completions/create">...</a>
+     */
+    @Value("${artemis.iris.models.gpt3_5.temperature}")
+    private float temperature;
+
+    /**
+     * The maximum number of tokens that the model should generate in its responses. This is a safety measure to prevent
+     * the model from returning too much data.
+     */
+    @Value("${artemis.iris.models.gpt3_5.max-generated-tokens}")
+    private int maxGeneratedTokens;
+
+    /**
+     * Up to four sequences of characters that the model should stop generating text at. This is a safety measure to
+     * prevent the model from generating too much text.
+     */
+    @Value("${artemis.iris.models.gpt3_5.stop-sequences}")
+    private String[] stopSequences;
+
+    public IrisGPT3_5Service(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    /**
+     * Prompts GPT-3.5 to generate a response to the conversation in the provided {@link IrisSession}.
+     *
+     * @param session The session to generate a response for.
+     * @return A {@link CompletableFuture} that will be completed with the response.
+     */
+    @Override
+    @Async
+    public CompletableFuture<String> getResponse(IrisSession session) {
```

Comment:
Just FYI for anyone else:
This method will likely change in the future:
- Return value will be a list of `IrisMessageContent` instead of String since the parsing of this is LLM dependant
- The parameter will likely change to a List of `IrisMessage` for database reasons

These aren't implemented yet, as they both kinda depend on the database stuff, which is a separate PR, so we will do this once we combine multiple PRs together.